### DISCUSSION TITLE
Ask for a import/export on prompt management page

### DISCUSSION THREAD

**Discussion by ShaunDeng** [langfuse/langfuse#2021] [Ideas]
### Describe the feature or potential improvement

May I ask for a new function to import/export the prompts on the prompt management page? Because I always need to migrate a list of prompt from one environment to another, the import/export will be useful to do such work. We may only focus on the latest version of prompt for importing/exporting.
<img width="1263" alt="Langfuse_prompt" src="https://github.com/langfuse/langfuse/assets/16474510/a2254893-a72b-4b92-a154-6de8249c8a93">

**Comment by marcklingen** [comment]
We were considering "copy to another project", what do you think about this?

**Reply by ShaunDeng** [reply]
Actually, I mean copying it to another environment, possibly located at a different host. Can the 'copy to another project' function be used across different environments?

**Reply by marcklingen** [reply]
Moving between hosts seems to be rather complicated to get right in a secure way. What drives the need of separating environments on different hosts?

**Reply by ShaunDeng** [reply]
Uh... We've deployed multiple environments for our developers (such as dev, tst, prod) on different hosts (in our on-premise cloud). When we release, we also need to migrate the prompt templates. We need to follow our company's compliance to store data from different environments separately.

**Reply by marcklingen** [reply]
Ok that makes sense. Currently this workflow is not really possible unless you manually copy/paste between the database tables of your instances. Alternatively, you could run a custom migration script to fetch the latest prompt versions via the get api and update your prod instance via the API as well.

**Comment by fancyweb** [comment]
@marcklingen How would you feel about simple JSON export and import endpoints?

My use case is making prompt synchronization between different environments (saas <-> self-hosted) easier. I'm currently using custom migrations but it's far from being ideal :sweat_smile:

**Reply by marcklingen** [reply]
Makes sense. Thank you for the clarification.

**Reply by rndninja** [reply]
Any updates on this? Are there any plans for JSON or any other format-based import and export endpoints?

**Reply by marcklingen** [reply]
Export/import is available via the SDKs/API, currently this is not yet available in the UI

**Reply by marcklingen** [reply]
Please upvote this thread as this helps to track overall interest for this feature

**Comment by anujsinha3** [comment]
Is this still in the pipeline? This will be a pretty useful feature similar to datasets export.

**Comment by hey-nicolasklein** [comment]
Same here, we run a dev and prod environment. Dev is only reachable in the internal network, while prod is accessible by our customers. Copying and pasting the prompts or running hacky sync scripts seems like the wrong way to do it. Prompts feel more like part of the code, like a "signed prompt version" that we want to carry over. Sure, we also don't sync our PostgreSQL between dev and prod, but it does not contain any logic relevant to the functionality of the application. Prompts are different from that and would need separate treatment.
